vizro-core/src/vizro/static/js/models/graph.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Decide whether a graph action should be triggered based on clickData / selectedData.
+ *
+ * Behavior:
+ * - If the graph does NOT support box selection (no "Box Select" in the modebar) -> propagate clickData.
+ * - If clickmode has been changed from "event+select" to something else and clickData is triggered -> propagate clickData.
+ * - If previous conditions are not met it means that we are in "event+select" mode and that the graph supports selection -> return selectedData.
+ *
+ * @param {Object|null} clickData     Plotly clickData from the graph.
+ * @param {Object|null} selectedData  Plotly selectedData from the graph.
+ * @param {Object|null} figure        Figure object used to inspect layout / clickmode.
+ * @param {string}      figureId      DOM id of the graph container used to check for modebar options.
+ * @returns {Object|null}             One of clickData, selectedData, or dash_clientside.no_update.
+ */
+function update_graph_actions_trigger_prop(
+  clickData,
+  selectedData,
+  figure,
+  figureId,
+) {
+  // Check whether graph supports selection by looking for "Box Select" option in the graph's modebar in chart's header.
+  const isGraphSelectable = !!document
+    .getElementById(figureId)
+    .querySelector('[data-title="Box Select"]');
+
+  // Extract the "clickmode" setting from the figure layout or default template.
+  // figure.layout.clickmode can be defined if user has explicitly set it with figure.update_layout(clickmode=...).
+  // figure.layout.template.layout.clickmode can be defined if user has overwritten the default template's clickmode.
```

Comment:
I think we can assume this is not possible since `event+select` is hardcoded into `dashboard_overrides` and not the `vizro_light/dark` themes themselves. So I would only check `figure.layout.clickmode` here.